bmx280.c

```diff
@@ -69,11 +69,28 @@
 // Value of REG_CHPID for BMP280 (Production)
 #define BMP280_ID2 0x58
 
+#if CONFIG_USE_I2C_NEW_DEVICE
+// I2C Master configuration structure (temporary)
+typedef struct i2c_config_t{
```

Comment:
I suggest you merge this into the main struct. I don't see a reason to have this be its own type.

Reply:
Thank you for your suggestion! I understand your perspective, I think the contents of this struct are not used again after the dev_handler is successfully created. By keeping it as a separate type and freeing it after use, we can save some memory.

Reply:
Not that a few bytes of memory really matters here, but then maybe stack memory is better for this? Does the information in this struct need to be used in bmx280_init?

Reply:
No, it is not used in the bmx280_init function, it is only used in the bmx280_create,bmx280_probe,bmx280_device_create function to set chip speed and address